### ISSUE TITLE
Components split: Composer install in zf components fails on dependency resolution

### ISSUE THREAD

**Issue by Xerkus** [zendframework/zf2#7537]
There are several issues currently with composer on split components.

`self.version` in cloned component is derived from first branch alias, "2.4-dev", as it seems and then converted to "2.4.9999999.9999999-dev".
Direct dependencies are resolved to dev-master but their own dependencies fail. Minimum-stability dev is required to complete installation.
Minimum stability amongst other things forces composer to pull dev-master for other libraries too. Eg for phpunit.

Component in dev env can only have dependencies at dev-master since composer is not aware of its own version. Unless composer.json is edited, of course.

I believe we have to get rid of `self.version` dependencies in components.
 
Example with freshly cloned zend-mvc: 
```
[Xerkus@laptop zend-mvc (master)]$ composer.phar install
Loading composer repositories with package information
Installing dependencies (including require-dev)
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Installation request for zendframework/zend-di == 2.4.9999999.9999999-dev -> satisfiable by zendframework/zend-di[2.4.x-dev].
    - zendframework/zend-di 2.4.x-dev requires zendframework/zend-code == 2.4.9999999.9999999-dev -> no matching package found.
  Problem 2
    - Installation request for zendframework/zend-http == 2.4.9999999.9999999-dev -> satisfiable by zendframework/zend-http[2.4.x-dev].
    - zendframework/zend-http 2.4.x-dev requires zendframework/zend-loader == 2.4.9999999.9999999-dev -> no matching package found.
  Problem 3
    - Installation request for zendframework/zend-serializer == 2.4.9999999.9999999-dev -> satisfiable by zendframework/zend-serializer[2.4.x-dev].
    - zendframework/zend-serializer 2.4.x-dev requires zendframework/zend-math == 2.4.9999999.9999999-dev -> no matching package found.
  Problem 4
    - Installation request for zendframework/zend-uri == 2.4.9999999.9999999-dev -> satisfiable by zendframework/zend-uri[2.4.x-dev].
    - zendframework/zend-uri 2.4.x-dev requires zendframework/zend-escaper == 2.4.9999999.9999999-dev -> no matching package found.
  Problem 5
    - Installation request for zendframework/zend-view == 2.4.9999999.9999999-dev -> satisfiable by zendframework/zend-view[2.4.x-dev].
    - zendframework/zend-view 2.4.x-dev requires zendframework/zend-loader == 2.4.9999999.9999999-dev -> no matching package found.

Potential causes:
 - A typo in the package name
 - The package is not available in a stable-enough version according to your minimum-stability setting
   see <https://groups.google.com/d/topic/composer-dev/_g3ASeIFlrc/discussion> for more details.

Read <https://getcomposer.org/doc/articles/troubleshooting.md> for further common problems.
```

**Comment by Xerkus** [comment]
Currently, installation works if "version" with precise release is added. Eg `"version": "2.4.2",`

**Comment by samsonasik** [comment]
should be works now? close-able?

**Comment by Xerkus** [comment]
Yes, it is resolved in 2.5 and upwards
